common/metrics.py

```
@@ -0,0 +1,209 @@
+from typing import TypeVar, Union, Dict, Any
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)